frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -68,67 +66,69 @@ const propTypes = {
+  if (isEmpty(param)) {
+    return {};
+  }
+  return typeof param === "string" ? { [key]: param } : { [key]: param.join(",") };
+};
+export const removeEmptyStings = (param, key) => (isEmpty(param) ? {} : { [key]: param });
+export const formatParams = ({
+  region = [],
+  compliance_code = [],
+  issue_date_after,
+  issue_date_before,
+  expiry_date_before,
+  expiry_date_after,
+  search,
+  major,
+  ...remainingParams
+}) => {
+  return {
+    ...joinOrRemove(region, "region"),
+    ...joinOrRemove(compliance_code, "compliance_code"),
+    ...removeEmptyStings(issue_date_after, "issue_date_after"),
+    ...removeEmptyStings(issue_date_before, "issue_date_before"),
+    ...removeEmptyStings(expiry_date_before, "expiry_date_before"),
+    ...removeEmptyStings(expiry_date_after, "expiry_date_after"),
+    ...removeEmptyStings(search, "search"),
+    ...removeEmptyStings(major, "major"),
+    ...remainingParams,
+  };
+};
 export class VarianceHomePage extends Component {
+  params = queryString.parse(this.props.location.search);
+
   constructor(props) {
     super(props);
+    const formatedParams = {
+      compliance_code: formatParamStringToArray(this.params.compliance_code),
+      region: formatParamStringToArray(this.params.region),
+      major: this.params.major,
+      search: this.params.search,
+      issue_date_after: this.params.issue_date_after,
+      issue_date_before: this.params.issue_date_before,
+      expiry_date_before: this.params.expiry_date_before,
+      expiry_date_after: this.params.expiry_date_after,
+    };
     this.handleVarianceSearchDebounced = debounce(this.handleVarianceSearch, 1000);
     this.state = {
       variancesLoaded: false,
-      params: {
-        page: Strings.DEFAULT_PAGE,
-        per_page: Strings.DEFAULT_PER_PAGE,
-        major: "",
-        region: [],
-        compliance_code: [],
-        issue_date_after: "",
-        issue_date_before: "",
-        expiry_date_before: "",
-        expiry_date_after: "",
-        search: "",
-      },
+      params: formatedParams,
     };
   }
 
-  state = {
-    variancesLoaded: false,
-    params: {
-      variance_application_status_code: [],
-      page: Strings.DEFAULT_PAGE,
-      per_page: Strings.DEFAULT_PER_PAGE,
-      major: "",
-      region: [],
-      compliance_code: [],
-      issue_date_after: "",
-      issue_date_before: "",
-      expiry_date_before: "",
-      expiry_date_after: "",
-      search: "",
-    },
-  };
-
   componentDidMount() {
     const params = this.props.location.search;
     const parsedParams = queryString.parse(params);
-    const {
-      page = this.state.params.page,
-      per_page = this.state.params.per_page,
-      type = this.state.params.type,
-    } = parsedParams;
+    const { page = this.state.params.page, per_page = this.state.params.per_page } = parsedParams;
     if (params) {
-      this.renderDataFromURL(params);
+      this.renderDataFromURL();
```

Comment:
Reason for removing this? Did it fix a bug? I imagine it was there for a reason

Reply:
I had put the 'params' in a previous pr because I had misread the logic. passing the params ended up being redundant.